frontend/dashboard/src/app/page.tsx

```diff
@@ -1,112 +1,153 @@
-import Image from "next/image";
+"use client";
+import {Button} from "@/components/ui/button";
+import {InputWithLabel} from "@/components/ui/inputWithLabel";
+import { SelectScrollable } from "@/components/ui/selectScrollable";
+import { College, Interest } from "@/const/dummyConst";
+import MultipleSelector from '@/components/ui/multiSelect';
+import { useForm, Controller } from "react-hook-form";
+import { z } from "zod";
+import {Option} from "@/types/option";
+import { useState } from 'react';
+
+type FormData = {
+  firstName: string;
+  lastName: string;
+  email: string;
+  college: string;
+  interests: Option[];
+};
+
+const formSchema = z.object({
+  firstName: z.string().min(2),
+  lastName: z.string().min(2),
+  email: z.string().email({
+    message: "Please enter your email"
+  }),
+  college: z.string().min(1),
+  interests: z.array(z.object({
+    label: z.string(),
+    value: z.string()
+  })).nonempty(),
+})
 
 export default function Home() {
+  const [formDataJson, setFormDataJson] = useState<string>('');
+  const {
+    control,
+    handleSubmit,
+    formState: { errors }
+} = useForm<FormData>();
+
+const onSubmit = (data: FormData) => {
+  formSchema.parse(data);
```

Comment:
you can pass the schema as a "resolver" to the useForm hook. 

useForm({
    resolver: zodResolver(formSchema),
  })

Reply:
This will automatically trigger and error / success before hitting the onSubmit